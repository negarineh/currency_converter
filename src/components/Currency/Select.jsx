import React from "react";

import { SelectCurrency } from "./styles";

const Select = ({ value, onChange, currencyList = [] }) => (
    <SelectCurrency value={value} onChange={onChange} name="currencyList">
        {currencyList.map(currency => (
            <option key={currency.id} value={currency.id}>
                {currency.currencyName}
            </option>
        ))}
    </SelectCurrency>
);

export default React.memo(Select);
