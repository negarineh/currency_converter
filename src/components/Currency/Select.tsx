import React from "react";

import SelectTypes from "./Select.types";
import { SelectCurrency } from "./styles";

const Select = ({ value, onChange, currencyList }: SelectTypes) => (
    <SelectCurrency value={value} onChange={onChange} name="currencyList">
        {currencyList.map(currency => (
            <option key={currency.id} value={currency.id}>
                {currency.currencyName}
            </option>
        ))}
    </SelectCurrency>
);

export default React.memo(Select);
