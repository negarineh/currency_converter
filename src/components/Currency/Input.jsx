import React from 'react';
import {
    CurrencyConverter,
    Input,
} from "./styles";

import Select from "./Select";

function InputCurrency({
    currencyList,
    inputValue,
    selectValue,
    inputChange,
    selectChange,
    inputRef,
}) {
    return (
        <CurrencyConverter>
            <Input
                type="number"
                value={inputValue}
                onChange={inputChange}
                ref={inputRef}
            />

            <Select
                value={selectValue}
                onChange={selectChange}
                currencyList={currencyList}
            />
        </CurrencyConverter>
    )
}

export default React.memo(InputCurrency);
