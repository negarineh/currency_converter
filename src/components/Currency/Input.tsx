import React from 'react';
import {
    CurrencyConverter,
    Input,
} from "./styles";

import Select from "./Select";
import InputTypes from './Input.types';

function InputCurrency({
    currencyList,
    inputValue,
    selectValue,
    inputChange,
    selectChange,
    inputRef,
}: InputTypes) {
    return (
        <CurrencyConverter>
            <Input
                type="number"
                value={inputValue}
                onChange={inputChange}
                ref={inputRef}
            />

            <Select
                value={selectValue}
                onChange={selectChange}
                currencyList={currencyList}
            />
        </CurrencyConverter>
    )
}

export default React.memo(InputCurrency);
