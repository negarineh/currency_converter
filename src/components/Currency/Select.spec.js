import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Select from './Select';
import { currencyList } from './Input.spec';

const value = 0;
const onChange = jest.fn();

test('render <Select />', () => {
    const container = renderer.create(<Select value={value} onChange={onChange} currencyList={currencyList} />).toJSON();
    expect(container.firstChild).toMatchSnapshot();
});

test('it applies styles', () => {
    const tree = renderer.create(<Select value={0} onChange={() => { }} currencyList={[]} />).toJSON();
    expect(tree).toHaveStyleRule('height', '3rem');
    expect(tree).toHaveStyleRule('width', '20rem');
    expect(tree).toHaveStyleRule('background-color', '#fff');
    expect(tree).not.toHaveStyleRule('opacity');
    expect(tree).toHaveStyleRule('border-radius', '5px');
    expect(tree).toHaveStyleRule('padding', '0.7rem');
});
