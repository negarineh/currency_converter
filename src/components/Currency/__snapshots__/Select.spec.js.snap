// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render <Input /> 1`] = `
.c2 {
  font-size: 0.9rem;
  height: 3rem;
  width: 20rem;
  background-color: #fff;
  border-radius: 5px;
  padding: 0.7rem;
}

.c2:focus {
  outline: none;
}

.c0 {
  margin-bottom: 1rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: inline-block;
}

.c1 {
  width: 14rem;
  padding: 0.7em;
  border: 1px solid #dbe2e8;
  color: #2e3d49;
  max-width: 50rem;
  margin-right: 2.5rem;
  outline: none;
  height: 3rem;
  font-size: 0.9rem;
  border-radius: 5px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: inline-block;
  margin: 1rem;
}

@media only screen and (max-width:600px) {
  .c2 {
    width: 14rem;
  }
}

<div
  className="c0"
>
  <input
    className="c1"
    onChange={[MockFunction]}
    type="number"
    value={0}
  />
  <select
    className="c2"
    name="currencyList"
    onChange={[MockFunction]}
    value={0}
  >
    <option
      value="AUD"
    >
      Australian Dollar
    </option>
    <option
      value="CAD"
    >
      Canadian Dollar
    </option>
  </select>
</div>
`;

exports[`render <Select /> 1`] = `undefined`;
